vault/resource_pki_backend_role.go

```diff
@@ -0,0 +1,380 @@
+			"max_ttl": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The maximum TTL of certificates issued using this role, provided as a duration.",
+			},
+			"allow_localhost": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if clients can request certificates for localhost as one of the requested common names.",
+			},
+			"allowed_domains": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "Policies to be set on tokens issued using this role.",
+			},
+			"allow_bare_domains": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if clients can request certificates matching the value of the actual domains themselves.",
+			},
+			"allow_subdomains": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if clients can request certificates with CNs that are subdomains of the CNs allowed by the other role options.",
+			},
+			"allow_glob_domains": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Allows names specified in allowed_domains to contain glob patterns.",
+			},
+			"allow_any_name": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if clients can request any CN.",
+			},
+			"enforce_hostnames": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if only valid host names are allowed for CNs, DNS SANs, and the host part of email addresses.",
+			},
+			"allow_ip_sans": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if clients can request IP Subject Alternative Names.",
+			},
+			"server_flag": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if certificates are flagged for server use.",
+			},
+			"client_flag": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Specifies if certificates are flagged for client use.",
+			},
+			"code_signing_flag": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if certificates are flagged for code signing use.",
+			},
+			"email_protection_flag": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "Specifies if certificates are flagged for email protection use.",
+			},
+			"key_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "rsa",
+				Description: "Specifies the type of key to generate for generated private keys. Currently, rsa and ec are supported.",
```

Comment:
Could we do a validation of this here? Here's an example of how that can be done: https://github.com/terraform-providers/terraform-provider-google/blob/ca6c39031f459828f32c69ef872ec8178e3996a5/google/data_source_google_compute_zones.go#L30

Reply:
Yep! Thanks for the pointer.